test/suite/test_timestamp08.py

```diff
@@ -0,0 +1,187 @@
+            'commit_timestamp=' + timestamp_str(3))
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(3))
+
+        self.session.begin_transaction()
+        c[2] = 2
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.timestamp_transaction(
+                'commit_timestamp=' + timestamp_str(2)),
+                '/older than oldest timestamp/')
+        self.session.rollback_transaction()
+
+        self.session.begin_transaction()
+        c[2] = 2
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.commit_transaction(
+                'commit_timestamp=' + timestamp_str(2)),
+                '/older than oldest timestamp/')
+
+        self.session.begin_transaction()
+        c[4] = 4
+        self.session.commit_transaction(
+            'commit_timestamp=' + timestamp_str(4))
+
+        # oldest timestamp <= stable timestamp and both oldest and stable
+        # timestamp should proceed forward.
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.conn.set_timestamp('oldest_timestamp=' +
+                timestamp_str(3) + ',stable_timestamp=' + timestamp_str(1)),
+                '/oldest timestamp must not be later than stable timestamp/')
+
+        # oldest timestamp is 3 at the moment, trying to set it to an earlier
+        # timestamp is a no-op
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1))
+        self.assertEqual(self.conn.query_timestamp('get=oldest'),
+            timestamp_str(3))
+
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(3) +
+            ',stable_timestamp=' + timestamp_str(3))
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(5))
+        # stable timestamp is 5 at the moment, trying to set it to an earlier
+        # timestamp is a no-op
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(4))
+        self.assertEqual(self.conn.query_timestamp('get=stable'),
+            timestamp_str(5))
+
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(5))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.conn.set_timestamp('oldest_timestamp=' +
+                timestamp_str(6)),
+                '/oldest timestamp must not be later than stable timestamp/')
+
+        # commit timestamp >= stable timestamp, stable timestamp is set to 5 at
+        # the moment.
+        # Also, when explicitly set, commit timestamp for a transaction can be
+        # earlier than the commit timestamp of an earlier transaction
+        self.session.begin_transaction()
+        c[2] = 2
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.commit_transaction(
+                'commit_timestamp=' + timestamp_str(2)),
+                '/older than oldest timestamp/')
```

Comment:
The comment discusses the stable timestamp and the code/error message use oldest timestamp.  One or the other is probably incorrect.

Reply:
good catch.
This isn't right, I wanted to check if commit ts >= stable ts and I haven't checked that correctly. I will have to put the test case in, as well as the code to catch it through timestamp_transaction and commit_transaction